src/test/java/org/javarosa/core/model/SelectChoiceTest.java

```diff
@@ -119,4 +123,103 @@ public void getChild_updates_whenChoicesAreFromRepeat() throws IOException {
         scenario.answer("/data/repeat[0]/special-property", "changed");
         assertThat(scenario.choicesOf("/data/select").get(0).getChild("special-property"), is("changed"));
     }
+
+    @Test
+    public void getChild_throwsException_whenCalledOnAChoiceFromStaticSelect() throws IOException {
+        Scenario scenario = Scenario.init("Static select", html(
+            head(
+                title("Static select"),
+                model(
+                    mainInstance(
+                        t("data id='static-select'",
+                            t("select"))))),
+            body(
+                select1("/data/select", item("one", "One"), item("two", "Two"))
+            )));
+
+        try {
+            scenario.choicesOf("/data/select").get(0).getChild("invalid-property");
+            fail();
+        } catch (IllegalStateException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void getAdditionalChildren_returnsChildren_whenChoicesAreFromSecondaryInstance() {
+        setUpSimpleReferenceManager(r("external-select-geojson.xml").getParent(), "file");
+
+        Scenario scenario = Scenario.init("external-select-geojson.xml");
+
+        Map<String, String> firstNodeChildren = scenario.choicesOf("/data/q").get(0).getAdditionalChildren();
+        assertThat(firstNodeChildren.keySet(), hasSize(3));
+        assertThat(firstNodeChildren, hasEntry("geometry", "0.5 102 0 0"));
+        assertThat(firstNodeChildren, hasEntry("id", "fs87b"));
+        assertThat(firstNodeChildren, hasEntry("foo", "bar"));
+
+        Map<String, String> secondNodeChildren = scenario.choicesOf("/data/q").get(1).getAdditionalChildren();
+        assertThat(secondNodeChildren.keySet(), hasSize(4));
+        assertThat(secondNodeChildren, hasEntry("geometry", "0.5 104 0 0"));
+        assertThat(secondNodeChildren, hasEntry("id", "67abie"));
+        assertThat(secondNodeChildren, hasEntry("foo", "quux"));
+        assertThat(secondNodeChildren, hasEntry("special-property", "special value"));
+    }
+
+    @Test
+    public void getChildren_updates_whenChoicesAreFromRepeat() throws IOException {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Select from repeat"),
+                model(
+                    mainInstance(
+                        t("data id='repeat-select'",
+                            t("repeat",
+                                t("value"),
+                                t("label"),
+                                t("special-property")),
+                            t("filter"),
+                            t("select"))))),
+            body(
+                repeat("/data/repeat",
+                    input("value"),
+                    input("label"),
+                    input("special-property")),
+                input("filter"),
+                select1Dynamic("/data/select", "../repeat")
```

Comment:
Wait... you can use a repeat as the "itemset" for a select?

Reply:
https://github.com/getodk/javarosa/pull/660/files#r832740208 😄